src/plugins/workspace/server/saved_objects/workspace_ui_settings_client_wrapper.ts

```diff
@@ -144,6 +148,61 @@ export class WorkspaceUiSettingsClientWrapper {
       return wrapperOptions.client.update(type, id, attributes, options);
     };
 
+    const deleteWithDefaultIndexPatternCheck = async (
```

Comment:
I'm a bit concern that we're over complicating this saved object wrapper, it's intended to be a wrapper for `config` type of object, but now we're doing an implicit cascading updating when a `index-pattern` is deleted.
Could you please elaborate why this change is necessary?

Reply:
You are right, we can keep the default index pattern validation logic in the ui setting PR when trying to delete index patterns. Then it will throw an error if a user tries to remove a default index pattern. The benefits of leaving the cascading updating logic were that the user doesn't need to call the UI setting separately. I'm open to this. After this change, the user will get an error when trying to uninstall sample data and the index pattern is a default index pattern. Do you think if we should remove the default index patterns ui setting before sample data uninstall?